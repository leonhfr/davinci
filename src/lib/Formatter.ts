// Internal.
import * as Types from '../types';

// Code.
export class Formatter {
  static formatResponse(input: any): any {
    if (typeof input !== 'object' || input === null) {
      return input;
    }

    const { page, pages, perpage, total } = input;
    return {
      ...input,
      page: Number(page),
      pages: Number(pages),
      perpage: Number(perpage),
      total: Number(total),
    };
  }

  static formatPhoto(input: any, properties?: Types.PhotoProperties): any {
    if (typeof input !== 'object' || input === null) {
      return input;
    }

    let { description, tags, views, latitude, longitude } = input;

    if (description && description._content) {
      description = description._content;
    }

    if (tags && typeof tags === 'string') {
      tags = tags.split(' ');
    }

    views = Number.isNaN(Number(views)) ? null : Number(views);
    latitude = Number.isNaN(Number(latitude)) ? null : Number(latitude);
    longitude = Number.isNaN(Number(longitude)) ? null : Number(longitude);

    return {
      ...input,
      description,
      tags,
      views,
      latitude,
      longitude,
      ...properties,
    };
  }
}
