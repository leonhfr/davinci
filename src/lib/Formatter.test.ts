// Internal.
import { Formatter } from './Formatter';

// Code.
describe('Formatter', () => {
  it('should be defined', () => {
    expect(Formatter).toBeDefined();
  });
});
