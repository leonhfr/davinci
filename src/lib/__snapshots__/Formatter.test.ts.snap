// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formatter formatPhoto should correctly format an API response 1`] = `
Object {
  "accuracy": 16,
  "context": 0,
  "description": "The Arc de Triomf was built as the entrance of the 1888 Barcelona Universal Exposition. It was designed by architect Josep Vilaseca i Casanovas in the Neo-Mudéjar style.",
  "farm": 5,
  "geo_is_contact": 0,
  "geo_is_family": 0,
  "geo_is_friend": 0,
  "geo_is_public": 1,
  "id": "32457360587",
  "isfamily": 0,
  "isfriend": 0,
  "ispublic": 1,
  "latitude": 41.390791,
  "longitude": 2.18114,
  "owner": "41710899@N08",
  "ownername": "jdf_92",
  "place_id": "iFsnXNZTUriu3usbkw",
  "secret": "69b27ceb18",
  "server": "4898",
  "tags": Array [
    "spain",
    "barcelona",
    "arcdetriomf",
    "1888barcelonaworldfair",
    "neomudéjar",
    "1888barcelonauniversalexposition",
    "catelonia",
  ],
  "title": "Arc de Triomf",
  "views": 16,
  "woeid": "20220092",
}
`;

exports[`Formatter formatPhoto should correctly format an API response 2`] = `
Object {
  "accuracy": 16,
  "context": 0,
  "description": undefined,
  "farm": 5,
  "geo_is_contact": 0,
  "geo_is_family": 0,
  "geo_is_friend": 0,
  "geo_is_public": 1,
  "id": "32457360587",
  "isfamily": 0,
  "isfriend": 0,
  "ispublic": 1,
  "latitude": 41.390791,
  "longitude": 2.18114,
  "owner": "41710899@N08",
  "ownername": "jdf_92",
  "place_id": "iFsnXNZTUriu3usbkw",
  "secret": "69b27ceb18",
  "server": "4898",
  "tags": undefined,
  "title": "Arc de Triomf",
  "views": 16,
  "woeid": "20220092",
}
`;

exports[`Formatter formatPhoto should correctly format an API response 3`] = `
Object {
  "accuracy": 16,
  "context": 0,
  "description": "The Arc de Triomf was built as the entrance of the 1888 Barcelona Universal Exposition. It was designed by architect Josep Vilaseca i Casanovas in the Neo-Mudéjar style.",
  "farm": 5,
  "geo_is_contact": 0,
  "geo_is_family": 0,
  "geo_is_friend": 0,
  "geo_is_public": 1,
  "id": "32457360587",
  "isfamily": 0,
  "isfriend": 0,
  "ispublic": 1,
  "latitude": null,
  "longitude": null,
  "owner": "41710899@N08",
  "ownername": "jdf_92",
  "place_id": "iFsnXNZTUriu3usbkw",
  "secret": "69b27ceb18",
  "server": "4898",
  "tags": Array [
    "spain",
    "barcelona",
    "arcdetriomf",
    "1888barcelonaworldfair",
    "neomudéjar",
    "1888barcelonauniversalexposition",
    "catelonia",
  ],
  "title": "Arc de Triomf",
  "views": null,
  "woeid": "20220092",
}
`;

exports[`Formatter formatPhoto should correctly pass properties 1`] = `
Object {
  "accuracy": 16,
  "context": 0,
  "description": "The Arc de Triomf was built as the entrance of the 1888 Barcelona Universal Exposition. It was designed by architect Josep Vilaseca i Casanovas in the Neo-Mudéjar style.",
  "farm": 5,
  "geo_is_contact": 0,
  "geo_is_family": 0,
  "geo_is_friend": 0,
  "geo_is_public": 1,
  "id": "32457360587",
  "inside": false,
  "isfamily": 0,
  "isfriend": 0,
  "ispublic": 1,
  "latitude": 41.390791,
  "longitude": 2.18114,
  "owner": "41710899@N08",
  "ownername": "jdf_92",
  "place_id": "iFsnXNZTUriu3usbkw",
  "secret": "69b27ceb18",
  "server": "4898",
  "tags": Array [
    "spain",
    "barcelona",
    "arcdetriomf",
    "1888barcelonaworldfair",
    "neomudéjar",
    "1888barcelonauniversalexposition",
    "catelonia",
  ],
  "title": "Arc de Triomf",
  "views": 16,
  "woeid": "20220092",
  "zoneId": "4ab7068b-6c6c-46d2-8009-1d7d1ab35a3b",
}
`;

exports[`Formatter formatResponse should correctly format an API response 1`] = `
Object {
  "page": 1,
  "pages": 25225,
  "perpage": 2,
  "photo": Array [],
  "total": 50450,
}
`;
